reactor-pool/src/main/java/reactor/pool/SimpleDequePool.java

```diff
@@ -829,7 +831,7 @@ public void onSubscribe(Subscription s) {
 
 		@Override
 		public void request(long l) {
-			if (Operators.validate(l)) {
+			if (upstream != null && Operators.validate(l)) {
```

Comment:
We know this can't be called unless the Subscriber was offered a reference of `this`. And that would only happen on line 828. One line above the value is assigned and on line 825 it's validated to be non-null. Therefore, perhaps we just need `assert upstream != null` and avoid the runtime null check?

Reply:
fixed